pallets/ethereum-xcm/src/lib.rs

```diff
@@ -127,12 +140,22 @@ pub mod pallet {
 				EthereumXcmTransaction::V1(v1_tx) =>  v1_tx.gas_limit.unique_saturated_into(),
 				EthereumXcmTransaction::V2(v2_tx) =>  v2_tx.gas_limit.unique_saturated_into()
 			}
-		}))]
+		}).saturating_add(T::DbWeight::get().reads(1)))]
 		pub fn transact(
 			origin: OriginFor<T>,
 			xcm_transaction: EthereumXcmTransaction,
 		) -> DispatchResultWithPostInfo {
 			let source = T::XcmEthereumOrigin::ensure_origin(origin)?;
+			ensure!(
+				!EthereumXcmSuspended::<T>::get(),
+				DispatchErrorWithPostInfo {
+					error: Error::<T>::EthereumXcmExecutionSuspended.into(),
+					post_info: PostDispatchInfo {
+						actual_weight: Some(T::DbWeight::get().reads(1)),
```

Comment:
This probably understates the cost slightly, although if we've had to suspend XCM->EVM then we probably don't care much.

Is the main point of this to avoid the normal double-charge mechanism we currently have?

Reply:
Why does it understate? If EthereumXcm is suspended, we only do 

` T::XcmEthereumOrigin::ensure_origin(origin)?;`

and

`EthereumXcmSuspended::<T>::get()`

Hence the reason why we charge only a db read. ensure_origin is a trivial conversion that shouldnt be irrelevant